packages/playground/data-liberation/src/entity-readers/WP_WXR_Sorted_Entity_Reader.php

```diff
@@ -0,0 +1,669 @@
+				mapped_id text DEFAULT NULL,
+				parent_id text DEFAULT NULL,
+				cursor_id text DEFAULT NULL,
+				sort_order int DEFAULT 1,
+				PRIMARY KEY  (id),
+				KEY session_id (session_id),
+				KEY entity_id (entity_id(%d)),
+				KEY parent_id (parent_id(%d)),
+				KEY cursor_id (cursor_id(%d))
+			) ' . $wpdb->get_charset_collate(),
+			self::get_table_name(),
+			$max_index_length,
+			$max_index_length,
+			$max_index_length
+		);
+
+		require_once ABSPATH . 'wp-admin/includes/upgrade.php';
+		// dbDelta is a function that creates the table.
+		dbDelta( $sql );
+	}
+
+	/**
+	 * Run by register_deactivation_hook or similar. It drops the table and
+	 * deletes the option.
+	 */
+	public static function delete_db() {
+		global $wpdb;
+
+		// Drop the table.
+		$wpdb->query(
+			$wpdb->prepare( 'DROP TABLE IF EXISTS %i', self::get_table_name() )
+		);
+	}
+
+	/**
+	 * Set the emit cursor flag. If true, the reader will emit the cursor ID
+	 * for each entity.
+	 *
+	 * @param bool $emit_cursor The emit cursor flag.
+	 */
+	public function set_emit_cursor( $emit_cursor ) {
+		$this->emit_cursor = $emit_cursor;
+	}
+
+	/**
+	 * Reset the class.
+	 */
+	public function reset() {
+		$this->current_session = null;
+	}
+
+	/**
+	 * Delete all rows for a given session ID.
+	 *
+	 * @param int $session_id The session ID to delete rows for.
+	 * @return int|false The number of rows deleted, or false on error.
+	 */
+	public function delete_session( $session_id = null ) {
+		global $wpdb;
+
+		return $wpdb->delete(
+			self::get_table_name(),
+			array( 'session_id' => $session_id ?? $this->current_session ),
+			array( '%d' )
+		);
+	}
+
+	/**
+	 * Add the next entity to the sorting table.
+	 *
+	 * @param string $entity_type The type of the entity.
+	 * @param array  $data The data to map.
+	 * @param mixed  $cursor_id The stream cursor ID.
+	 */
+	public function add_next_entity( $entity = null ) {
+		global $wpdb;
+
+		// We're done if all the entities are processed
+		if ( ! $this->valid() ) {
+			return false;
+		}
+
+		$entity      = $entity ?? $this->current();
+		$data        = $entity->get_data();
+		$entity_type = $entity->get_type();
+
+		// Do not need to be mapped, skip it.
+		if ( ! array_key_exists( $entity_type, self::ENTITY_TYPES ) ) {
+			// Advance to next entity.
+			$this->next();
+
+			return true;
+		}
+
+		// Default sort order is 1.
+		$sort_order = 1;
+		$cursor_id  = $this->get_reentrancy_cursor();
+
+		// The new entity to be added to the table.
+		$new_entity = array(
+			'session_id'  => $this->current_session,
+			'entity_type' => self::ENTITY_TYPES[ $entity_type ],
+			'entity_id'   => null,
+			'mapped_id'   => null,
+			'parent_id'   => null,
+			'cursor_id'   => $cursor_id,
+			'sort_order'  => 1,
+		);
+
+		// Get the ID of the entity.
+		$entity_id      = (string) $data[ self::ENTITY_TYPES_ID[ $entity_type ] ];
+		$parent_id_type = null;
+		$check_existing = true;
+
+		// Map the parent ID if the entity has one.
```

Comment:
Why are we mapping IDs while sorting? Mapping seems like a separate problem we'll need to handle for every entity source, not just the WXR one.

Reply:
I may be confused here. It seems like "mapping parent ID" refers to assigning the `parent` value to `$new_entity` and not the "reading a post with one ID but inserting it with another parent ID." Also, `entity` refers to the thing we import, `$new_entity` refers to a row in the database table we use for sorting.

As I'm read, I find myself slowing down frequently to understand the ideas behind the words. I'd like to ask you to revisit the terminology used in this PR, especially in comments and variable and function names. Let's make the words more precise than they need to be. My goal is to make all the concepts unambiguous so that a junior WordPress dev could read it without getting confused about any part of the codebase. It will make the next review much easier for me.